File: library/src/main/java/me/proxer/library/api/ucp/DeleteFavoriteEndpoint.java
package me.proxer.library.api.ucp;

import lombok.experimental.Accessors;
import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;
import org.jetbrains.annotations.NotNull;

/**
 * Endpoint for unmarking an entry as favorite.
 *
 * @author Ruben Gees
 */
@Accessors(fluent = true)
public final class DeleteFavoriteEndpoint implements Endpoint {

    private final InternalApi internalApi;

    private final String id;

    DeleteFavoriteEndpoint(@NotNull final InternalApi internalApi, @NotNull final String id) {
        this.internalApi = internalApi;
        this.id = id;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    @NotNull
    public ProxerCall<Void> build() {
        return internalApi.deleteFavorite(id);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/ucp/DeleteFavoriteEndpoint.kt
package me.proxer.library.api.ucp

import me.proxer.library.ProxerCall
import me.proxer.library.api.Endpoint

/**
 * Endpoint for unmarking an entry as favorite.
 *
 * @author Ruben Gees
 */
class DeleteFavoriteEndpoint internal constructor(
    private val internalApi: InternalApi,
    private val id: String
) : Endpoint<Unit> {

    override fun build(): ProxerCall<Unit> {
        return internalApi.deleteFavorite(id)
    }
}
